client/src/containers/DashboardContainer/DashboardContainer.js

```diff
@@ -35,28 +42,37 @@ const DashboardContainer = (props) => {
 
   const [breakpoint, setBreakpoint] = useState('lg');
   const [cols, setCols] = useState(null);
-  const [isEdit, setIsEdit] = useState(false);
-  const [oldName, setOldName] = useState('');
-  const [name, setName] = useState(null);
-  const [oldDescription, setOldDescription] = useState('');
-  const [description, setDescription] = useState(null);
-
+  const [viewDashboardMode, setViewDashboardMode] = useState('default');
   const [isModalVisible, setIsModalVisible] = useState(false);
```

Comment:
as a suggestion. We can use useReducer hook to work with all properties.

Reply:
i will refactor this in a another branch